### PR TITLE
Only notarize CI builds; timeout after 90 min

### PR DESCRIPTION
**Pull request by cadsit**

### CODE REVIEW

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -503,28 +501,26 @@ timestamps {
...
-                                    if (notarize_ios)
-                                        pkgs = pkgs + xiPackages
                                     withCredentials([string(credentialsId: 'codesign_keychain_pw', variable: 'KEYCHAIN_PASS'), string(credentialsId: 'team_id', variable: 'TEAM_ID'), string(credentialsId: 'application_id', variable: 'APP_ID'), string(credentialsId: 'installer_id', variable: 'INSTALL_ID'), usernamePassword(credentialsId: 'apple_account', passwordVariable: 'APPLE_PASS', usernameVariable: 'APPLE_ACCOUNT')]) {
                                         sh (returnStatus: true, script: "security create-keychain -p ${env.KEYCHAIN_PASS} login.keychain") // needed to repopulate the keychain
                                         sh ("security unlock-keychain -p ${env.KEYCHAIN_PASS} login.keychain")
-                                        sh ("python release-scripts/sign_and_notarize.py -a ${env.APP_ID} -i ${env.INSTALL_ID} -u ${env.APPLE_ACCOUNT} -p ${env.APPLE_PASS} -t ${env.TEAM_ID} -d package/notarized -e ${entitlements} -k login.keychain " + pkgs.flatten().join(" "))
+                                        timeout(time: 90, unit: 'MINUTES') {
```
Comment (rolfbjarne) [inline jenkins/Jenkinsfile]:
```suggestion
                                        timeout (time: 90, unit: 'MINUTES') {
```

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -503,28 +501,26 @@ timestamps {
...
+                                          sh ("python release-scripts/sign_and_notarize.py -a ${env.APP_ID} -i ${env.INSTALL_ID} -u ${env.APPLE_ACCOUNT} -p ${env.APPLE_PASS} -t ${env.TEAM_ID} -d package/notarized -e ${entitlements} -k login.keychain " + pkgs.flatten().join(" "))
```
Comment (rolfbjarne) [inline jenkins/Jenkinsfile]:
```suggestion
                                          sh ("python release-scripts/sign_and_notarize.py -a ${env.APP_ID} -i ${env.INSTALL_ID} -u ${env.APPLE_ACCOUNT} -p ${env.APPLE_PASS} -t ${env.TEAM_ID} -d package/notarized -e ${entitlements} -k login.keychain " + pkgs.flatten ().join (" "))
```

File: jenkins/Jenkinsfile (hunk 1 above)
Comment (mandel-macaque) [reply jenkins/Jenkinsfile]: Other timeout calls do have the space, other methods no. But lets be consistent.

**Comment by rolfbjarne** [comment]
@cadsit I had to revert this, because it failed the build: https://jenkins.internalx.com/blue/organizations/jenkins/macios/detail/master/935/pipeline/57

Notarization fails with:

> Scripts not permitted to use staticMethod org.codehaus.groovy.runtime.DefaultGroovyMethods flatten java.lang.Object[]

which eventually causes 


    MicrosoftAzureStorage - Container name: wrench
    MicrosoftAzureStorage - share name: 
    MicrosoftAzureStorage - File path: package/notarized/*
    MicrosoftAzureStorage - Virtual path: jenkins/master/cd396d63ccee7455d409355f61b438afbf410b36/935/
    MicrosoftAzureStorage - Exclude path: 
    MicrosoftAzureStorage - Uploading files to Microsoft Azure
    MicrosoftAzureStorage - Need uploaded/archived file count = 0
    MicrosoftAzureStorage - Uploaded/archived file count = 0
    MicrosoftAzureStorage - Failed to find any build artifacts to upload to Azure Storage 
    Verify the list of files to upload and that the Ant glob syntax is correct
    ERROR: MicrosoftAzureStorage - Error occurred while uploading to Azure - bosstoragemirror
    java.io.IOException: MicrosoftAzureStorage - Failed to find any build artifacts to upload to Azure Storage 
    Verify the list of files to upload and that the Ant glob syntax is correct
    	at com.microsoftopentechnologies.windowsazurestorage.WAStoragePublisher.perform(WAStoragePublisher.java:430)
    	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:80)
    	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
    	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
    	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    	at java.lang.Thread.run(Thread.java:748)

so there are two bugs:

* Calling `pkgs.flatten()` is not allowed
* Trying to upload 0 notarized packages (which ends up failing)

**Comment by cadsit** [comment]
Ah -- I should have been able to catch that. I've been bitten by that a few times before.

There's a subtle difference between list types in groovy and by not initializing `pkgs = []`, I think we ended up with a type that has not yet been approved through the glories of Jenkins Catalina-esque security system. The annoying part, however, is that I don't have the method available to approve (I think because of exception handling).

**Pull request merged**